lib/Conversion/TorchToTosa/TorchBackendLegalToTosaCustom.cpp

```diff
@@ -0,0 +1,220 @@
+          int64_t intConstAttr;
+          if (matchPattern(intOperand, m_TorchConstantInt(&intConstAttr))) {
+            return tosa::getConstTensor<int64_t>(rewriter, backendLegalOp,
+                                                 intConstAttr, {})
+                .value();
+          }
+          return nullptr;
+        })
+        .Case<Torch::ConstantFloatOp>([&](Operation *floatOperand) -> Value {
+          double floatConstAttr;
+          if (matchPattern(floatOperand,
+                           m_TorchConstantFloat(&floatConstAttr))) {
+            return tosa::getConstTensor<float>(rewriter, backendLegalOp,
+                                               floatConstAttr, {})
+                .value();
+          }
+          return nullptr;
+        })
+        .Case<Torch::ConstantNoneOp>([&](Operation *noneOperand) -> Value {
+          auto noneCustomOp = rewriter.create<tosa::CustomOp>(
+              backendLegalOp->getLoc(),
+              RankedTensorType::get({}, rewriter.getIntegerType(1)),
+              rewriter.getStringAttr("constant.none"),
+              rewriter.getStringAttr("torch_mlir"), rewriter.getStringAttr(""),
+              ValueRange{});
+          return noneCustomOp.getResult(0);
+        })
+        // TODO Add support for converting "torch.constant.number"
+        .Case<Torch::ConstantNumberOp>(
+            [&](Operation *strOperand) -> Value { return nullptr; })
+        .Case<Torch::ConstantStrOp>([&](Operation *strOperand) -> Value {
+          std::string strConstAttr;
+          if (matchPattern(strOperand, m_TorchConstantStr(strConstAttr))) {
+            auto strCustomOp = rewriter.create<tosa::CustomOp>(
+                backendLegalOp->getLoc(),
+                RankedTensorType::get({}, rewriter.getIntegerType(8)),
+                rewriter.getStringAttr("constant.str"),
+                rewriter.getStringAttr("torch_mlir"),
+                rewriter.getStringAttr(""), ValueRange{});
+            return strCustomOp.getResult(0);
+          }
+          return nullptr;
+        })
+        .Case<Torch::PrimListConstructOp>(
+            [&](Operation *intListConstructOperand) -> Value {
+              SmallVector<int64_t> intConstListAttr;
+              if (matchPattern(intListConstructOperand,
+                               m_TorchListOfConstantInts(intConstListAttr))) {
+                return tosa::getConstTensor<int64_t>(
+                           rewriter, backendLegalOp, intConstListAttr,
+                           {static_cast<int64_t>(intConstListAttr.size())})
+                    .value();
+              }
+              return nullptr;
+            })
+        .Default([&](Operation *defaultOperand) { return inputOperand; });
+  }
+
+  LogicalResult
+  matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const final {
+
+    if (customOps.contains(op->getName().getStringRef())) {
```

Comment:
nit: return early to avoid indentation

https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code